govcd/network_pool.go

```diff
@@ -102,3 +102,271 @@ func (vcdClient *VCDClient) GetNetworkPoolByName(name string) (*NetworkPool, err
+	managers, err := vcdClient.QueryNsxtManagerByName(managerName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(managers) == 0 {
+		return nil, fmt.Errorf("no manager '%s' found", managerName)
+	}
+	if len(managers) > 1 {
+		return nil, fmt.Errorf("more than one manager '%s' found", managerName)
+	}
+	manager := managers[0]
+
+	managerId := "urn:vcloud:nsxtmanager:" + extractUuid(managers[0].HREF)
+	transportZones, err := vcdClient.GetAllNsxtTransportZones(managerId, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving transport zones for manager '%s': %s", manager.Name, err)
+	}
+	var transportZone *types.TransportZone
+	for _, tz := range transportZones {
+		// if the transport zone name was empty, we take the first available
+		// otherwise, we take the wanted transport zone
+		if (transportZoneName == "" && !tz.AlreadyImported) || tz.Name == transportZoneName {
+			transportZone = tz
+			break
+		}
+	}
+
+	if transportZone == nil {
+		if transportZoneName == "" {
+			return nil, fmt.Errorf("no unimported transport zone was found")
+		}
+		return nil, fmt.Errorf("transport zone '%s' not found", transportZoneName)
+	}
+
+	if transportZone.AlreadyImported {
+		return nil, fmt.Errorf("transport zone '%s' is already imported", transportZone.Name)
+	}
+
+	// Note: in this type of network pool, the managing owner is the NSX-T manager
+	managingOwner := types.OpenApiReference{
+		Name: manager.Name,
+		ID:   managerId,
+	}
+	var config = &types.NetworkPool{
+		Name:             name,
+		Description:      description,
+		PoolType:         types.NetworkPoolGeneveType,
+		ManagingOwnerRef: managingOwner,
+		Backing: types.NetworkPoolBacking{
+			TransportZoneRef: types.OpenApiReference{
+				ID:   transportZone.Id,
+				Name: transportZone.Name,
+			},
+			ProviderRef: managingOwner,
+		},
+	}
+	return vcdClient.CreateNetworkPool(config)
+}
+
+// CreateNetworkPoolPortGroup creates a network pool of PORTGROUP_BACKED type
+// The function retrieves the given vCenter and corresponding port group names
+// If the port group name is empty, the first available will be used
+func (vcdClient *VCDClient) CreateNetworkPoolPortGroup(name, description, vCenterName, portgroupName string) (*NetworkPool, error) {
+	vCenter, err := vcdClient.GetVCenterByName(vCenterName)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving vCenter '%s': %s", vCenterName, err)
+	}
+	var params = make(url.Values)
+	params.Set("virtualCenter.id", vCenter.VSphereVCenter.VcId)
+	portgroups, err := vcdClient.GetAllVcenterImportableDvpgs(params)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving portgroups for vCenter '%s': %s", vCenterName, err)
+	}
+	var portgroup *VcenterImportableDvpg
+	for _, pg := range portgroups {
+		// If the port group name was empty, we take the first available
+		// otherwise, we take the wanted one
```

Comment:
Similar query to the Geneve create method - it doesn't look that UI supports such randomness and one must always pick (check) a particular port group

Reply:
https://github.com/dataclouder/terraform-provider-vcd/blob/network-pools/website/docs/r/network_pool.html.markdown#example-usage-5---nameless-backing-retrieval